test/functional/change_stream.test.js

```diff
@@ -18,17 +18,27 @@ chai.use(require('chai-subset'));
  * Triggers a fake resumable error on a change stream
  *
  * @param {ChangeStream} changeStream
- * @param {function} onCursorClosed callback when cursor closed due this error
+ * @param {Function} onCursorClosed callback when cursor closed due this error
+ * @param {number} [delay] optional delay before triggering error
  */
-function triggerResumableError(changeStream, onCursorClosed) {
-  const closeCursor = changeStream.cursor.close;
+function triggerResumableError(changeStream, onCursorClosed, delay) {
+  let cbCalled = false;
+  const closeCursor = changeStream.cursor.close.bind(changeStream.cursor);
```

Comment:
I don't know if this is any cleaner, but we have [patterns](https://github.com/mongodb/node-mongodb-native/blob/master/test/functional/scram_sha_256.test.js#L182-L193) for using `sinon` to do this type of patching